janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/BufferUtil.java

```diff
@@ -57,6 +64,39 @@ public static StaticBuffer getIntBuffer(int[] ids) {
         return StaticArrayBuffer.of(arr);
     }
 
+    /**
+     * If string is not 8 bytes, then we write it as it is. Otherwise, we write a special
+     * marker at the end, so that when JanusGraph sees the buffer, it knows it should
+     * reinterpret the byte buffer as string buffer. See {@link IDManager#getKeyID(StaticBuffer)}
+     * for more details.
+     *
+     * @param s
+     * @return
+     */
+    public static StaticBuffer getStringBuffer(String s) {
+        if (StringUtils.isEmpty(s) || !StringEncoding.isAsciiString(s)) {
+            throw new IllegalArgumentException("value must be non-empty ASCII string but received: " + s);
+        }
+        ByteBuffer buffer;
+        if (s.length() == longSize) {
+            buffer = ByteBuffer.allocate(longSize + byteSize);
+        } else {
+            buffer = ByteBuffer.allocate(s.length());
+        }
+        for (int i = 0; i < s.length(); i++) {
+            int c = s.charAt(i);
+            assert c <= 127;
+            byte b = (byte)c;
+            if (i+1==s.length()) b |= 0x80; //End marker
```

Comment:
Why we need this `End marker`? Not sure I understand the difference between this `End marker` and `STOP_MASK`.

Reply:
Great observation! Here, we don't necessarily need this `0x80` as an end marker because, upon read time, we know the length of the buffer. This is more to keep it consistent with `StringEncoding#writeAsciiString` where an end marker is necessary.

The additional benefit of using an end marker (which effectively changes the first bit to 1) is that we can use it for corruption detection in the future.

The variable names and magic constants are indeed confusing. I'll rename them.